pkg/doc/verifiable/credential.go

```diff
@@ -220,7 +220,10 @@ type embeddedCompositeIssuer struct {
 type CredentialDecoder func(dataJSON []byte, credential *Credential) error
 
 // CredentialTemplate defines a factory method to create new Credential template.
-type CredentialTemplate func() *Credential
+// It's possible that Verifiable Credential could be not relevant to the Template's data model.
+// It can be determined by Template checking the types of the Verifiable Credential.
+// In this case error is returned.
+type CredentialTemplate func(credential *RawCredential) (*Credential, error)
```

Comment:
I don't understand why this template function needs the whole "raw credential" - it would appear it only needs the `@context` and the `type` properties as inputs, as evidenced [here](https://github.com/hyperledger/aries-framework-go/pull/227/files#diff-3f2d2cc11d10cd0a96b1574e700ae6dfR46)

Reply:
In my opinion, this is a more flexible approach - provide the basic VC to make a decision if the template is capable of extending it. Anyway, this approach can be changed, see the thread with @troyronda above (https://github.com/hyperledger/aries-framework-go/pull/227#discussion_r324332470).